src/saveEventsFile.m

```diff
@@ -289,32 +325,55 @@ function printHeaderExtraColumns(logFile)
 
         logFile = checklLogFile('fields', logFile, iEvent, cfg);
 
-        onset = logFile(iEvent).onset;
-        duration = logFile(iEvent).duration;
-        trial_type = logFile(iEvent).trial_type;
-
-        % we skip events with onset or duration that are empty, nan or char
-        if any(cell2mat(cellfun(@isnan, {onset duration}, 'UniformOutput', false))) || ...
-                any(cellfun(@ischar, {onset duration})) || ...
-                any(isempty({onset duration}))
-
-            warning('saveEventsFile:emptyEvent', ...
-                    '\nSkipping saving this event.\n onset: %s \n duration: %s\n', ...
-                    onset, ...
-                    duration);
-
-        else
-
-            printData(logFile(1).fileID, onset, cfg);
-            printData(logFile(1).fileID, duration, cfg);
-            printData(logFile(1).fileID, trial_type, cfg);
+        % if this is _events file, we skip events with onset or duration 
+        % that are empty, nan or char. 
+        if logFile(1,1).isStim==0 
+            
+            onset = logFile(iEvent).onset;
+            duration = logFile(iEvent).duration;
+            trial_type = logFile(iEvent).trial_type;
+
+            if any(cell2mat(cellfun(@isnan, {onset duration}, 'UniformOutput', false))) || ...
+               any(cellfun(@ischar, {onset duration})) || ...
+               any(isempty({onset duration})) 
+
+                warning('saveEventsFile:emptyEvent', ...
+                        '\nSkipping saving this event.\n onset: %s \n duration: %s\n', ...
+                        onset, ...
+                        duration);
+            else
+                printData(logFile(1).fileID, onset, cfg);
+                printData(logFile(1).fileID, duration, cfg);
+                printData(logFile(1).fileID, trial_type, cfg);
+                
+                printExtraColumns(logFile, iEvent, cfg);
 
-            printExtraColumns(logFile, iEvent, cfg);
+                fprintf(logFile(1).fileID, '\n');
+                fprintf(1, '\n');
 
-            fprintf(logFile(1).fileID, '\n');
-            fprintf(1, '\n');
+            end
+            
+        % if this is _stim file, we skip missing events (i.e. events where 
+        % all extra columns have NO values)
+        elseif logFile(1,1).isStim==1
+            
+            skipEvent = false; 
```

Comment:
I REALLY like your approach of setting this to false and then checking for if we want to skip and then printing.

I think this can actually be applied to the whole sub-function and we could refactor things that way.

```
skipEvent = false; 
% check if must skip
if isStim
elseif ~isStim
end

if skipEvent
 Warning('generic message that we are skipping saving');  % <-------- I guess we can tweak if to print different stuff for stim and events

else 
% print to file
  if ~isStim
                printData(logFile(1).fileID, onset, cfg);
                printData(logFile(1).fileID, duration, cfg);
                printData(logFile(1).fileID, trial_type, cfg);
   end
   printExtraColumns(logFile, iEvent, cfg);

   fprintf(logFile(1).fileID, '\n');
   printf(1, '\n');
end
```

Reply:
great idea, looks much better that way!